addons/account_payment/models/account_payment.py

```diff
@@ -132,7 +132,10 @@ def action_post(self):
         )
         # creating the transaction require to access data on payment providers, not always accessible to users
         # able to create payments
-        transactions = payments_need_tx.sudo()._create_payment_transaction()
+        extra_create_values = {}
+        if self._context.get('active_model') == 'account.move':
+            extra_create_values['invoice_ids'] = self._context.get('active_ids')
+        transactions = payments_need_tx.sudo()._create_payment_transaction(**extra_create_values)
```

Comment:
I'm not a fan of passing that through the `extra_create_values`, which is typically there only to allow easy overrides. Instead, prepare the value for `invoice_ids` directly inside `_prepare_payment_transaction_vals`.

Reply:
Done